### PR TITLE
Mock client

### PR DESCRIPTION
**Pull request by tamccall**

### CODE REVIEW

**Comment by tamccall** [comment]
@nukosuke currious if you have any thoughts on this. Can't quite get the go 1.8 build to work with gomock. I could remove it as a build step, but i wanted to check with you first on that.

**Comment by nukosuke** [comment]
@tamccall

TravisCI of gomock does not have go1.8 for build matrix. 
https://github.com/golang/mock/blob/master/.travis.yml

I don’t know if they explicitly purged it from support versions. But, it means that go1.8 is not tested and no guarantee to work.
I think it reasonable that we also remove it from support.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 language: go
 
 go:
-  - 1.8
   - 1.9
   - 1.10.x
   - 1.11.x
```
Comment (nukosuke) [inline .travis.yml]: Would you add 1.12.x instead?

**Comment by nukosuke** [comment]
OK, maybe I understood the overview of gomock.
It generates mock code from interface by `go generate`. All we need to do is adding interface definitions right?

**Comment by tamccall** [comment]
yeah once you have installed mockgen you should be able to generate the mock client with `go generate ./...` you should do that anytime you add a method to that API interface.

**Comment by nukosuke** [comment]
I see. Thanks for making that easy to understand!

**Pull request merged**